### PR TITLE
[Minor] Enable pybind in mac build

### PR DESCRIPTION
**Pull request by raymondxyang**
Mac pybind lib build was previously broken by changes. Enable it to avoid it occurs again.

### CODE REVIEW

**Comment by raymondxyang** [comment]
Send out PR later

**Comment by snnn** [comment]
--enable_pybind  and --build_wheel, you only need one

**Comment by jywu-msft** [comment]
> --enable_pybind  and --build_wheel, you only need one

prefer --build_wheel

**Comment by raymondxyang** [comment]
yep.. just removed the redundant one

File: tools/ci_build/github/azure-pipelines/mac-ci-pipeline.yml [hunk 1]
```diff
@@ -5,8 +5,9 @@ jobs:
   steps:
     - template: templates/set-test-data-variables-step.yml
     - script: |
+        sudo python3 -m pip install numpy==1.15.0
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/mac-ci-pipeline.yml]: Is this needed ?

numpy==1.15.0 is only for building redistributable packages.  This is not a packaging pipeline.

Comment (raymondxyang) [reply tools/ci_build/github/azure-pipelines/mac-ci-pipeline.yml]: This PR basically is to make sure the change did not break packaging pipeline. if changes cannot compile with old version, this place will be updated and at least we'll be informed about it and make corresponding changes in release pipelines.

**Pull request merged**